Client/Extensions/Rewards/BraveRewardsExtensions.swift

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveRewards
+import BraveShared
+
+extension BraveRewards {
+    /// Whether or not rewards is enabled
+    @objc public var isEnabled: Bool {
+        get {
+            ledger.isWalletCreated && ledger.isEnabled && ads.isEnabled
+        }
+        set {
+            willChangeValue(for: \.isEnabled)
+            Preferences.Rewards.rewardsToggledOnce.value = true
+            createWalletIfNeeded { [weak self] in
+                guard let self = self else { return }
+                self.ledger.isEnabled = newValue
+                self.ledger.isAutoContributeEnabled = newValue
+                self.ads.isEnabled = newValue
+                self.didChangeValue(for: \.isEnabled)
+            }
+        }
+    }
+    
+    public func createWalletIfNeeded(_ completion: (() -> Void)? = nil) {
+        if ledger.isWalletCreated {
+            completion?()
+            return
+        }
+        if isCreatingWallet {
+            // completion block will be hit by previous call
+            return
+        }
+        isCreatingWallet = true
+        ledger.createWalletAndFetchDetails { [weak self] success in
+            self?.isCreatingWallet = false
+            completion?()
+        }
+    }
+    
+    public var isCreatingWallet: Bool {
```

Comment:
Curious why dont we have this property on the framework side instead of associating an object at runtime ?

Reply:
Cause adding code to brave-core is a pain and this was on a tight deadline :) But really it was just added as a safety precaution, starting in brave-core 1.18 I think this will go away due to Rewards Everywhere